### ISSUE TITLE
Fix Request: false positive in AvoidUnguardedAssignmentsToNonFinalFields using afterPropertiesSet

### ISSUE THREAD

**Issue by jborgers** [jborgers/PMD-jPinpoint-rules#154]
If we only do unguarded assignment to the non-final field in a Spring method afterPropertiesSet, this is a false positive because Spring takes care of thread-safety on Spring beans methods.

example:
 ```
@Component 
class Foo {
  private Consumer consumer;

   public void afterPropertiesSet() {
        consumer = client.buildConsumer(consumerConfig, paymentProcessor);
        consumer.startConsuming();
    }
}
```
